include/dpp/export.h

```diff
@@ -116,11 +117,13 @@ extern bool DPP_EXPORT validate_configuration();
 }
 
 #ifndef _WIN32
-	#define SOCKET int
+	#ifndef SOCKET
```

Comment:
this is a bad idea, what youre saying is if the user defines SOCKET, then they break dpp and wsock2.h silently by changing the type in the DPP headers. (as opposed to it complaining about macro redefinition)

Reply:
Agreed. If you want to change the define, it could be a type alias instead something like `dpp::socket`

Reply:
That would be the best way but didnt know where to put it, so I just edited the macro